client/components/CreateAccount/CreateAnonymousAccount.jsx

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+    Button,
+    Header,
+    Loader,
+    Modal,
+    Popup,
+    Segment
+} from 'semantic-ui-react';
+import { logIn } from '@client/actions';
+import Session from '@client/util/session';
+import anonymousUsername from './anonymousUsername';
+import './CreateAnonymousAccount.css';
+
+class CreateAnonymousAccount extends Component {
+    constructor(props) {
+        super(props);
+
+        const from =
+            this.props.location.state && this.props.location.state.from;
+
+        this.state = {
+            from,
+            username: ''
+        };
+
+        this.onClick = this.onClick.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+    }
+
+    componentDidMount() {
+        this.generateUnusedAnonymousUsername();
+    }
+
+    async generateUnusedAnonymousUsername() {
+        const username = anonymousUsername();
+        try {
+            await fetch(`/api/auth/signup/usernames/${username}/exists`);
+        } catch (error) {
+            if (error) {
+                return await this.generateUnusedAnonymousUsername();
```

Comment:
Maybe we should confirm that this error is that the username exists? If it's an unrelated server error, I could see us getting caught in a re-try loop.